docs/lib/datastore/fragments/flutter/data-access/query-predicate-snippet.md

```diff
@@ -0,0 +1,9 @@
+```dart
+TBD
+```
+
+<amplify-callout warning>
+
+**Note:** when constructing predicates, static `QueryField` instances such as `Post.RATING` do not own any information about the model to which the field belongs. In order to avoid any ambiguity between field names which are used across multiple models, it is recommended to construct a custom instance of `QueryField` in the form of `QueryField.field("{model-name}.{field-name}")` (i.e. `field("post.rating")`).
+
+</amplify-callout>
```

Comment:
I believe you are actively addressing this in codegen, right?
```suggestion
```

Reply:
cc: @fjnoyp for this question.

Reply:
Jameson is correct and this section should be removed for Dart.  The codegen models will generate the id field as "post.id". 

As mentioned earlier today, the iOS side does not require this so for now we will add code that parses out the model name on iOS.